src/main/java/poolsheen/Storage.java

```diff
@@ -0,0 +1,127 @@
+package poolsheen;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.function.Predicate;
+import java.util.Scanner;
+
+import poolsheen.task.Task;
+import poolsheen.task.Event;
+import poolsheen.task.Deadline;
+import poolsheen.task.ToDo;
+
+/**
+ * Represents everything that deals with reading from and writing to files.
+ */
+public class Storage {
+    /** The file path of the save file that the Storage object will work with. */
+    private String filePath;
+
+    /** The file object representing the save file which the Storage object will work with. */
+    private File saveFile;
+
+    /**
+     * A public constructor to initialise a Storage object.
+     *
+     * @param filePath The path of the file which the Storage object will be interacting with.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.saveFile = new File(this.filePath);
+        //Automatically creates a new save file
+        try {
+            if (this.saveFile.createNewFile()) {
+                System.out.println("New Save File created");
+            } else {
+                System.out.println("File already exists");
+            }
+        } catch (IOException e) {
+            System.out.println("Error when constructing Storage object.");
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Returns an ArrayList of Tasks by parsing the contents of the save file.
+     *
+     * @throws IOException Thrown if there is an error with parsing the file.
+     */
+    public ArrayList<Task> load() throws PoolsheenException {
+        ArrayList<Task> res = new ArrayList<>(100);
```

Comment:
Perhaps the variable `res` could use a more precise name, like `tasks` or `listOfTasks`. Just a thought!